emulator/src/main/java/dk/dbc/search/solrdocstore/emulator/BibliographicResource.java

```diff
@@ -0,0 +1,66 @@
+package dk.dbc.search.solrdocstore.emulator;
+
+import org.eclipse.microprofile.metrics.annotation.Timed;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ejb.EJBException;
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+import java.net.URI;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+
+@Stateless
+@Path("/")
+@Produces("application/json")
+public class BibliographicResource {
+    private static final Logger log = LoggerFactory.getLogger(BibliographicResource.class);
+    private static final ObjectMapper O = new ObjectMapper();
+
+    @Inject
+    private Config config;
+
+    @POST
+    @Path("bibliographic")
+    @Consumes({MediaType.APPLICATION_JSON})
+    @Produces({MediaType.APPLICATION_JSON})
+    @Timed
+    public Response forwardBibliographicKeys(@QueryParam("skipQueue") @DefaultValue("false") boolean skipQueue,
+                                             String jsonContent) throws Exception
+    {
+        UriBuilder solrDocStoreUrlBuilder = config.getSolrDocStoreUriBuilder();
+        URI uri = solrDocStoreUrlBuilder
+                .path("api/bibliographic")
+                .queryParam("skipQueue", skipQueue)
+                .build();
+        log.debug("Forwarding json: {} to url: {}", jsonContent, uri.getHost() + uri.getPath());
+        Response resp = ClientBuilder.newClient()
+                .target(uri)
+                .request(MediaType.APPLICATION_JSON_TYPE)
+                .post(Entity.json(jsonContent));
+        String content = resp.readEntity(String.class);
+        if (resp.getStatusInfo().equals(Response.Status.OK) &&
+                resp.getMediaType().equals(MediaType.APPLICATION_JSON_TYPE)) {
+            // things are looking OK, return whatever solr-doc-store returned.
+            Object o = O.readValue(content, Object.class);
+            return Response.ok(o).build();
```

Comment:
Er det ikke lidt at gå over åen efter vand her? Jeg tror du kan hælde din streng, hvis den er velformed JSON, direkte i "Response.ok"? Så slipper du også for at have en ObjectMapper... :-)

Reply:
meningen var at kaste en exception hvis det ikke kan parses/er json. Men jeg ved faktisk ikke om det kan forekomme når getMediaType siger det er json?

Reply:
Jeg vil tro at din "Response.ok" nok skal kaste, hvis den ikke er tilfreds?

Reply:
ok, så. Jeg har fjernet objectmapperen.